src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -444,7 +444,13 @@ void VM_Version::initialize() {
   }
 
   if (UseSVE > 0) {
-    _initial_sve_vector_length = get_current_sve_vector_length();
+    int vl = get_current_sve_vector_length();
+    if (vl < 0) {
+      warning("Unable to get SVE vector length on this system. Disabling SVE.");
+      FLAG_SET_DEFAULT(UseSVE, 0);
+    } else {
+      _initial_sve_vector_length = vl;
+    }
```

Comment:
I think we need to disable SVE in `VM_Version::get_os_cpu_info` in `src/hotspot/os_cpu/linux_aarch64/vm_version_linux_aarch64.cpp`:

```
  if (auxv2 & HWCAP2_SVE2) _features |= CPU_SVE2;
  if (auxv2 & HWCAP2_SVEBITPERM) _features |= CPU_SVEBITPERM;

  if (prctl(PR_SVE_GET_VL) == -1) {
    _features &= ~CPU_SVE;
  }
```

Reply:
Aha. I think the other way around: we need to do more checks right newar the `_initial_sve_vector_length` inits. Hold on...

Reply:
See new commit.